### PR TITLE
Migrate docs to paradox

### PR DESCRIPTION
**Pull request by jonas**
A best effort attempt at lifting Sphinx docs to Markdown.
See it in action here: https://jonas.github.io/akka-http/

Some outstanding issues which are not done in this PR:

- @@snip for most scala directives do not work and this PR (`git grep FIXME@@snip | wc -l` shows 154)
- Cross-references to akka/akka documentation (`git grep 'FIXME: unresolved link reference' | wc -l` shows 12)
- Replacing `@vars@`

Note:
 - I didn't delete the `*.rst` files. Let me know if I should squash that into the conversion commit.
 - Also let me know if I need to prefix commits with `=doc` or whether you want me to squash everything. I kept the commits separate to improve reviewability.

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TES

**Comment by ktoso** [comment]
OK TO TEST

File: docs/src/test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala [hunk 1]
```diff
@@ -65,9 +65,9 @@ class HttpClientExampleSpec extends WordSpec with Matchers with CompileOnlySpec
 
     // while API remains the same to consume dataBytes, now they're in memory already:
     val transformedData: Future[ExamplePerson] =
-      strictEntity flatMap { e ⇒
+      strictEntity flatMap { e =>
```
Comment (ktoso) [inline docs/src/test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala]: good

**Comment by jonas** [comment]
@ktoso Should I squash and repush?

**Comment by ktoso** [comment]
We can squash from the github ui, so you don't need to :) Thanks for the great work so far, I'll skim it through before merging but likely should be merged today. Thanks!

**Comment by ktoso** [comment]
Skimmed through, looks promising/good!

**Comment by ktoso** [comment]
Thanks a lot, I'll also mark the new ticket as triaged.

**Comment by ktoso** [comment]
Resolves https://github.com/akka/akka-http/issues/1

**Pull request merged**